### PR TITLE
Refresh synthetics when user settings change

### PR DESCRIPTION
**Pull request by tgodzik**
This enables us to quickly toggle the decorations in the clients:

metals-sample-1605780598336

@ckipp01 I think since the settings are managed by the clients, we need to do the actual toggle command in the client. This will be most useful in VS Code and not sure if it will be helpful in other editors. Especially without decorations it doesn't change anything.

### CODE REVIEW

File: docs/editors/overview.md [hunk 1]
```diff
@@ -418,6 +427,12 @@ Language Server Protocol (LSP). These extensions are not necessary for Metals to
 function but they improve the user experience. To learn more about Metals
 extensions, see [integrating a new editor](../integrations/new-editor.md).
 
+## Implicit decorations 
+
+**✅**: Additional information inferred from the code can be show within the code using additional virtual text.
```
Comment (ckipp01) [inline docs/editors/overview.md:432]:
```suggestion
**✅**: Additional information inferred from the code can be show within the code using virtual text.
```
Super nitpicky, I think we can just remove the additional here.

File: docs/editors/overview.md [hunk 2]
```diff
@@ -418,6 +427,12 @@ Language Server Protocol (LSP). These extensions are not necessary for Metals to
...
+
+**Shown in hover**: Additional information inferred from the code can be show when hovering over a specific line. That hover only shows the current line.
```
Comment (ckipp01) [inline docs/editors/overview.md:434]:
```suggestion
**Shown in hover**: Additional information inferred from the code can be show when hovering over a symbol. That hover only shows information for the current line the symbol is one.
```
Not sure if you can actually just hover anywhere on the in other letters to trigger this or if it actually has to be on something. 🤔 take it or leave it.

**Review by ckipp01 — APPROVED** [review]
Just a couple nits on wording in the docs, but other than that, LGTM!

**Comment by ckipp01** [comment]
> @ckipp01 I think since the settings are managed by the clients, we need to do the actual toggle command in the client. This will be most useful in VS Code and not sure if it will be helpful in other editors. Especially without decorations it doesn't change anything.

For sure, this looks fantastic in VS Code 😍

File: docs/editors/overview.md (hunk 1 above)
Comment (tgodzik) [reply docs/editors/overview.md:432]: Fixed!

File: docs/editors/overview.md (hunk 2 above)
Comment (tgodzik) [reply docs/editors/overview.md]: This is anywhere on the line, since it seemed natural as we show the whole line. I will leave it with `line`

Comment (ckipp01) [reply docs/editors/overview.md]: 👍

**Pull request merged**